cdap-docs/application-templates/source/etl/templates/sources/batch/stream.rst

```diff
@@ -6,29 +6,85 @@
+  
+.. rubric:: Use Case
+
+The source is used whenever you need to read from a stream in batch. For example,
+you may want to read from a stream every hour, perform some data cleansing, then write
+the cleansed data for that hour as avro files.
+
+.. rubric:: Properties
+
+**name:** Name of the stream. Must be a valid stream name. If the stream does not exist,
+it will be created.
     
-**Duration:** Size of the time window to read with each run of the pipeline. The format is
+**duration:** Size of the time window to read with each run of the pipeline. The format is
 expected to be a number followed by a 's', 'm', 'h', or 'd' specifying the time unit, with
 's' for seconds, 'm' for minutes, 'h' for hours, and 'd' for days. For example, a value of
 '5m' means each run of the pipeline will read 5 minutes of events from the stream.
 
-**Delay:** Optional delay for reading stream events. The value must be of the same format
+**delay:** Optional delay for reading stream events. The value must be of the same format
 as the duration value. For example, a duration of '5m' and a delay of '10m' means each run
 of the pipeline will read events from 15 minutes before its logical start time to 10
 minutes before its logical start time. The default value is 0.
 
-**Format:** Optional format of the stream. Any format supported by CDAP is also supported.
+**format:** Optional format of the stream. Any format supported by CDAP is also supported.
 For example, a value of 'csv' will attempt to parse stream events as comma separated
 values. If no format is given, event bodies will be treated as bytes, resulting in a three
 field schema: 'ts' of type long, 'headers' of type map of string to string, and 'body' of
 type bytes.
 
-**Schema:** Optional schema for the body of stream events. Schema is used in conjunction
+**schema:** Optional schema for the body of stream events. Schema is used in conjunction
 with format to parse stream events. Some formats like the avro format require schema,
 while others do not. The schema given is for the body of the stream, so the final schema
 of records output by the source will contain an additional field named 'ts' for the
 timestamp and a field named 'headers' for the headers as as the first and second fields of
 the schema.
+
+.. rubric:: Example
+
+::
+
+  {
+    "name": "Stream",
+    "properties": {
+      "name": "accesslogs",
+      "duration": "10m",
+      "delay": "5m",
+      "format": "clf"
+    }
+  }
+
+The example reads from a stream named 'accesslogs'. It reads ten minutes worth of data from
+the stream, starting 15 minutes before the logical start time of the run to 5 minutes before
+the logical start time of the run. For example, if the pipeline was scheduled to run at 10:00am,
+the source will read data from 9:45am to 9:55am. The stream contents will be parsed as 
+'clf' (Combined Log Format), which will output records with the following schema::
```

Comment:
Should we explain how the calculation works a bit more? How does 15 minutes come out of these settings?

The last line could be reduced to: "which outputs records with this schema:"